### DISCUSSION TITLE
How do I remove all filters from nvtree?

### DISCUSSION THREAD

**Discussion by voodoojunk** [NvChad/NvChad#2486] [Q&A]
Hi 

I want to remove all filters from nvtree.
Currently `plugins/configs/nvimtree.lua` has this:
```
  filters = {
    dotfiles = false,
    exclude = { vim.fn.stdpath "config" .. "/lua/custom" },
  },
```
I tried to override that in `custom/configs/overrides.lua` in the nvimtree block with this:
```
  filters = {
    dotfiles = true,
    exclude = {}
}
```
This does not work however.
I suspect this is due to vim.tbl_deep_extend treating the empty table as a map instead of a list.
How do I handle this?

**Comment by siduck** [comment]
@voodoojunk it works for me tho 

https://github.com/NvChad/NvChad/assets/59060246/2ab5f0f3-052d-46da-bd35-91bdc3ac579c

**Reply by voodoojunk** [reply]
I'm sorry - I don't understand your video example. You didn't open nvimtree in the the video. How do you know if it works?
I have a stock cofiguration and all I added was the override (like you did), and when I open nvimtree I do not see my custom dir

**Reply by siduck** [reply]
@voodoojunk i printed the nvimtree config with/without the option  which is clear that it works here

**Comment by flatoy** [comment]
In normal mode hit <kbd>Shift</kbd> + <kbd>i</kbd>, or <kbd>Shift</kbd> + <kbd>h</kbd>, which will show hidden files. Changing the filters didn't work for me when I wanted to see `.env` files.

**Reply by siduck** [reply]
@indiehackor make a video so i could believe it!

**Comment by alewis001** [comment]
@voodoojunk (taking "and when I open nvimtree I do not see my custom dir" as the issue you're experiencing...) There is a `.gitignore` file which, if I remember correctly, ignores the custom dir and as such becomes hidden in nvim-tree. I used this config in my custom plugins lua file in order to show the ignored files/dirs; maybe it will help you...

```
   {
    "nvim-tree/nvim-tree.lua",
    opts = {
      live_filter = {
        always_show_folders = true,
      },
      git = {
        enable = true,
      },
      renderer = {
        highlight_git = true,
        group_empty = true,
        icons = {
          show = {
            git = true,
          }
        }
      },
      view = {
        width = 50
      }
    }
  },
```

I mean, I know it says to exclude that dir in the property but I can see the dir in nvim-tree...

![Screenshot 2023-11-13 at 13 55 01](https://github.com/NvChad/NvChad/assets/3482446/624a0aca-a65d-4c0f-989d-1b5ed839c95b)

...or maybe I'm misunderstanding the issue.

**Reply by voodoojunk** [reply]
OK - I think I found a weird workaround. I can get the same view (with the grayed out custom dir with the "ignored" symbol) by just setting exclude to a list with an empty string:
```
  filters = {
    dotfiles = true,
    exclude = {""}
  }, 
```

**Reply by siduck** [reply]
@voodoojunk  just go to nvimtree's  setup func and delete the exclude key from that table, simple 
![image](https://github.com/NvChad/NvChad/assets/59060246/d83449f0-a22c-440a-aaaa-fa608e7b7118)
